detection_rules/rule.py

```diff
@@ -481,9 +481,10 @@ class HistoryWindowStart:
     def validate(self, meta: RuleMeta) -> None:
         """Validates terms in new_terms_fields are valid ECS schema."""
 
-        super(NewTermsRuleData, self).validate_query(meta)
-        feature_min_stack = '8.4.0'
-        feature_min_stack_extended_fields = '8.6.0'
+        kql_validator = KQLValidator(self.query)
```

Comment:
```suggestion
        assert self.language == 'kquery', \ 
            "new terms field only supports the kquery language" 
        kql_validator = KQLValidator(self.query)
```

Reply:
Confirmed `mixins.MarshmallowDataclassMixin.from_dict` will fail with schema validation errors if `language` is eql. Attempted to add that as the rule type as well and if failed too.

Reply:
due to FilterLanguages = Literal["kuery", "lucene"]